python/chronos/src/bigdl/chronos/data/tsdataset.py

```diff
@@ -774,6 +774,16 @@ def to_torch_data_loader(self,
             invalidInputError(not self._has_generate_agg_feature,
                               "Currently to_torch_data_loader does not support "
                               "'gen_global_feature' and 'gen_rolling_feature' methods.")
+
+            if isinstance(self.horizon, int):
+                need_dflen = self.lookback + self.horizon
+            else:
+                need_dflen = self.lookback + max(self.horizon)
+            if len(self.df) < need_dflen:
+                invalidInputError(False,
+                                  "The length of the dataset must be larger than "
+                                  "'lookback' plus 'horizon'! ")
```

Comment:
f"The length of the dataset must be larger than the sum of lookback and horizon, while get lookback+horizon={lookback+horizon} and length of dataset={len(self.df)}"